#########################################
#SRD129 16S - OTU table


#Purpose: Generate an OTU table using files generated by mothur.

#Files needed:
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared

#The output files from mothur needed in R for this section and subsequent sections, aside from fasta files, are text files that can be saved as csv for ease of use in R.

#Load library package
library(tidyverse)

#######################################################################

#To start creating OTU table, edit taxonomy file
#Using bash copy file to have a different name
#cp SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy ./SRD129Flu.outsingletons.abund.taxonomy.csv


taxonomy <- read_tsv("./SRD129Flu.outsingletons.abund.taxonomy.csv") #Import this csv file from working directory using "read.csv" function
taxonomy$Taxonomy <- gsub('*\\(.*?\\) *', '', taxonomy$Taxonomy) #Substitute (100) and variations of that with nothing ('') from the Taxonomy column
taxonomy[1:6,3] #Show column number 3, rows 1 through 6 in 'taxonomy' dataframe
write.csv(taxonomy, file = "SRD129abundsingleton4277taxonomy.csv") #Write 'taxonomy' into a csv file and open the csv file in a spreadsheet editor to remove the size and numbered rows

#Using bash copy file to have a different name
#cp SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared SRD129Flu.outsingletons.abund.subsample.shared.csv

shared <- read_tsv("./SRD129Flu.outsingletons.abund.subsample.shared.csv")
head(shared) #Check on the first part of 'shared' dataframe
shared[1:6,1]
shared <- t(shared) #Transpose 'shared'
head(shared)
shared[1:6,1]
write.csv(shared, file = 'SRD129abundsingleton4277shared.csv') #Open this csv file in a spreadsheet editor and remove the "V*", "label", and "numOtus" rows

#Read the revised SRD129abundsingleton4277shared.csv file
shared <- read.csv("./SRD129abundsingleton4277shared.csv", stringsAsFactors = FALSE)
colnames(shared) [1] <- "OTU" #Rename first column of 'shared' to "OTU"
taxonomy <- read.csv("./SRD129abundsingleton4277taxonomy.csv")
OTUtable <- merge(shared, taxonomy, by.x ="OTU", by.y = "OTU") #Merge 'shared' and 'taxonomy' objects by OTU
head(OTUtable)
nrow(OTUtable) #Count number of rows in 'OTUtable' #1393
ncol(OTUtable) #Count number of columns in 'OTUtable' #157
write.csv(OTUtable, file= "SRD129abundsingleton4277OTUtable.csv")
#Open this csv file in a spreadsheet editor and remove the first row (numbered rows) and "Size" column #columns

#Check OTU table
OTUtable <-read.csv("./SRD129abundsingleton4277OTUtable.csv", stringsAsFactors = FALSE)
head(OTUtable) #Check the first part of 'OTUtable' to make sure it looks ok
